configs/tables/activatedDeactivatedColleaguesTableConfig.js

```diff
@@ -4,6 +4,17 @@
  */
 const activatedDeactivatedColleaguesTable = [{
```

Comment:
Just an idea: I saw you created a complex code structure in this folder, but it does not have a schema. 
You could store a JSdoc @typedef somewhere in a central place, and then reference it using @type. This would allow your IDE to validate the structure and highlight if there are typos, and you can use autocompletion in the IDE. 

Here you would add: 
```js
/**
 * @type {Array.<TableField>}
 */
```

And them in central place you would add, and them import it: 
```js
/**
 * @typedef {Object} TableField
 *
 * @property {string} label - The label of the field.
 * @property {Object} [headingAttributes] - The attributes for the heading of the field.
 * @property {string} [headingAttributes.role] - The role of the heading.
 * @property {string} [headingAttributes.tabindex] - The tab index of the heading.
 * @property {Object} [headingInnerTag] - The inner tag of the heading.
 * @property {string} [headingInnerTag.tagName] - The name of the inner tag.
 * @property {Object} [headingInnerTag.htmlAttributes] - The attributes for the inner tag of the heading.
 * @property {string} [headingInnerTag.htmlAttributes.id] - The id of the inner tag.
 * @property {string} [headingInnerTag.htmlAttributes.class] - The class of the inner tag.
 * @property {Array.<string>} value - The value of the field.
 * @property {string} [html] - The HTML tag for the field.
 * @property {Object} [htmlAttributes] - The HTML attributes for the field.
 * @property {string} [htmlAttributes.class] - The CSS class of the field.
 * @property {string} [htmlAttributes.type] - The type of the field, if it is a button.
 * @property {string} [htmlAttributes.id] - The ID of the field, if any.
 * @property {string} [htmlAttributes.data-license-id] - The ID of the license, if any.
 * @property {string} [htmlAttributes.data-modal-id] - The ID of the modal to open, if any.
 * @property {string} [htmlAttributes.data-contactid] - The ID of the contact, if any.
 * @property {string} [htmlAttributes.data-next-action-class] - The class for the next action, if any.
 * @property {Array.<string>} [htmlAttributes.data-user-firstname] - The first name of the user, if any.
 * @property {Array.<string>} [htmlAttributes.data-user-lastname] - The last name of the user, if any.
 * @property {Array.<string>} [htmlAttributes.data-user-jobtitle] - The job title of the user, if any.
 * @property {Array.<string>} [htmlAttributes.data-user-email] - The email of the user, if any.
 * @property {Array.<string>} [htmlAttributes.data-user-phone] - The phone number of the user, if any.
 * @property {string} [htmlAttributes.data-new-productuserid] - The ID of the new product user, if any.
 */
```

I generated this docs using chatGPT4. (prompt: create jsdoc @typedef for the following code that represents fields in a table: {...}, {...}  )

Reply:
Thank you, I agree we should have some jsdoc for better readability.
Thought on cleaning it up to below, as for the attributes it is pretty obvious and is kind of free field so is just not restricted to what is mentioned and is kind of confusing to mention them

`/**
 * @typedef {Object} TableField
 *
 * @property {string} label - The label of the table heading field.
 * @property {Object} [headingAttributes] - The attributes for the tables heading of the field,
 *  if any.
 * @property {Object} [headingInnerTag] - The inner HTML tag of the heading tag th, if any.
 * @property {string} [headingInnerTag.tagName] - The tag name of inner HTML tag of the heading.
 * @property {Object} [headingInnerTag.htmlAttributes] -
 * The attributes for the inner HTML tag of the heading.
 * @property {Array.<string>} value - The table division value.
 * @property {string} [html] - The HTML tag for the table division value, default is td.
 * @property {Object} [htmlAttributes] - The HTML attributes for table division value tag.
 * @property {Function} processValueMethod - The method used to process the field's value, if any.
 */`